File: mycollab-dao/src/main/java/com/esofthead/mycollab/core/arguments/BitSearchField.java
/**
 * This file is part of mycollab-dao.
 *
 * mycollab-dao is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-dao is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-dao.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.core.arguments;

import com.esofthead.mycollab.core.utils.BeanUtility;

/**
 * 
 * @author MyCollab Ltd.
 * @since 1.0
 */
public class BitSearchField extends NumberSearchField {
	private static final long serialVersionUID = 1L;

	public static final BitSearchField TRUE = new BitSearchField(AND, 1);
	public static final BitSearchField FALSE = new BitSearchField(AND, 0);

	public BitSearchField() {
		this(AND, 0);
	}

	public BitSearchField(Number value) {
		this(SearchField.AND, value, EQUAL);
	}

	public BitSearchField(String oper, Number value) {
		this(oper, value, EQUAL);
	}

	public BitSearchField(Number value, String compareOperator) {
		this(SearchField.AND, value, compareOperator);
	}

	public BitSearchField(String oper, Number value, String compareOperator) {
		super(oper, value, compareOperator);
	}

	public String toString() {
		return BeanUtility.printBeanObj(this);
	}
}

File: mycollab-dao/src/main/java/com/mycollab/db/arguments/BitSearchField.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.db.arguments

import com.mycollab.core.utils.BeanUtility

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
class BitSearchField @JvmOverloads constructor(oper: String = SearchField.AND, value: Int = 0, compareOperator: String = NumberSearchField.EQUAL) : NumberSearchField(oper, value, compareOperator) {

    override fun toString(): String {
        return BeanUtility.printBeanObj(this)
    }

    companion object {
        private val serialVersionUID = 1L

        @JvmField val TRUE = BitSearchField(SearchField.AND, 1)
        @JvmField val FALSE = BitSearchField(SearchField.AND, 0)
    }
}
